circleci.sh

```diff
@@ -89,7 +89,7 @@ coverage() {
 
     # rebuild dmd with coverage enabled
     # use the just build dmd as host compiler this time
-    mv src/dmd src/host_dmd
+    mv generated/dmd src/host_dmd
```

Comment:
> So, if this PR goes further all the scripts in the dlang repo must be updated. Is this hard to accomplish?

Well, as mentioned this only applies if we go with `dmd` as module name (for `ddmd` there's no conflict).
Moreover as you have seen the auto-tester clones all three repos, so you would need to go in three steps:

1) DMD: Create a new Makefile target that exposed `generated/dmd`
2) For all depending repos: send PRs to switch from `src/dmd` to `generated/dmd`
3) Merge this PR

Btw another workaround would be to use `source` instead of `src` (you can then softly deprecate src/dmd`). As @ZombineDev pointed out `source` is the DUB default folder for source code files.